pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,305 @@
+	configurationSetName := awsclient.String(mg.GetAnnotations()[meta.AnnotationKeyExternalName])
+
+	// update DeliveryOptions (PutConfigurationSetDeliveryOptions)
+	if !isUpToDateDeliveryOptions(cr, e.ConfigurationSetObservation) {
+		deliveryOptionsInput := &svcsdk.PutConfigurationSetDeliveryOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			SendingPoolName:      cr.Spec.ForProvider.DeliveryOptions.SendingPoolName,
+			TlsPolicy:            cr.Spec.ForProvider.DeliveryOptions.TLSPolicy,
+		}
+		if _, err := e.client.PutConfigurationSetDeliveryOptionsWithContext(ctx, deliveryOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update ReputationOptions (PutConfigurationSetReputationOptions)
+	if !isUpToDateReputationOptions(cr, e.ConfigurationSetObservation) {
+		reputationOptionsInput := &svcsdk.PutConfigurationSetReputationOptionsInput{
+			ConfigurationSetName:     configurationSetName,
+			ReputationMetricsEnabled: cr.Spec.ForProvider.ReputationOptions.ReputationMetricsEnabled,
+		}
+		if _, err := e.client.PutConfigurationSetReputationOptionsWithContext(ctx, reputationOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update SuppressionOptions (PutConfigurationSetSuppressionOptions)
+	var suppresssedReasons []*string
+	if !isUpToDateSuppressionOptions(cr, e.ConfigurationSetObservation) {
+		suppresssedReasons = cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons
+		supressOptionsInput := &svcsdk.PutConfigurationSetSuppressionOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			SuppressedReasons:    suppresssedReasons,
+		}
+		if _, err := e.client.PutConfigurationSetSuppressionOptionsWithContext(ctx, supressOptionsInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update TrackingOptions (PutConfigurationSetTrackingOptions)
+	if !isUpToDateTrackingOptions(cr, e.ConfigurationSetObservation) {
+		trackingOptionInput := &svcsdk.PutConfigurationSetTrackingOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			CustomRedirectDomain: cr.Spec.ForProvider.TrackingOptions.CustomRedirectDomain,
+		}
+		if _, err := e.client.PutConfigurationSetTrackingOptionsWithContext(ctx, trackingOptionInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// update SendingOptions (PutConfigurationSetSendingOptions)
+	if !isUpToDateSendingOptions(cr, e.ConfigurationSetObservation) {
+		sendingOptionInput := &svcsdk.PutConfigurationSetSendingOptionsInput{
+			ConfigurationSetName: configurationSetName,
+			SendingEnabled:       cr.Spec.ForProvider.SendingOptions.SendingEnabled,
+		}
+		if _, err := e.client.PutConfigurationSetSendingOptionsWithContext(ctx, sendingOptionInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
```

Comment:
There should be a different error message for each `Wrap` to help identify where the error came from. `"update failed"` is too generic here.

Reply:
resolved